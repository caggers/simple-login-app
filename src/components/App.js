import React, { Component } from 'react';
import Login from './Login';
import ErrorBoundary from './ErrorBoundary';

class App extends Component {
  constructor() {
    super();
    this.state = {
      isAuthenticated: false,
    };
  }

  authenticateUser = () => {
    this.setState({ isAuthenticated: true });
  };

  render() {

    return (
      <div>
        {this.state.isAuthenticated ? (
          <div className="welcome-message">Welcome Back </div>
        ) : (
          <ErrorBoundary>
            <Login authenticateUser={this.authenticateUser} className="login" />
          </ErrorBoundary>
        )}
      </div>
    );
  }
}

export default App;
