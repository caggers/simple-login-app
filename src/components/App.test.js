import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  const app = shallow(<App />);
  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('initialises with an isAuthenticated in `state`', () => {
    expect(app.state().isAuthenticated).toEqual(false);
  });

  it('updates the `state` of isAuthenticated', () => {
    app.instance().authenticateUser();
    expect(app.state().isAuthenticated).toEqual(true);
    app.setState({ isAuthenticated: false });
  });

  it('renders the login component conditionally', () => {
    app.setState({ isAuthenticated: false });
    expect(app.find('.login')).toExist();
  });

  it('renders the welcome component conditionally', () => {
    app.setState({ isAuthenticated: true });
    expect(app.find('.welcome-message')).toExist();
  });
});
