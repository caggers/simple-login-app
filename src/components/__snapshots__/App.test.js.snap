// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <ErrorBoundary>
    <Login
      authenticateUser={[Function]}
      className="login"
    />
  </ErrorBoundary>
</div>
`;
